src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.ProjectAndCompilationWithAnalyzers.cs

```diff
@@ -1,23 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-namespace Microsoft.CodeAnalysis.Diagnostics;
-
-internal partial class DiagnosticAnalyzerService
-{
-    private partial class DiagnosticIncrementalAnalyzer
-    {
-        private sealed class ProjectAndCompilationWithAnalyzers
-        {
-            public Project Project { get; }
-            public CompilationWithAnalyzersPair? CompilationWithAnalyzers { get; }
```

Comment:
instead of a weakref to one of these, we have a CWT from a Project to the last CompPair we computed for it.